dask/optimize.py

```diff
@@ -102,6 +115,8 @@ def fuse(dsk, keys=None):
         val = dsk[child]
         while chain:
             parent = chain.pop()
+            dependencies[parent].update(dependencies.pop(child))
+            dependencies[parent].remove(child)
```

Comment:
`dependencies` will still contain items that were fused into other keys.  Perhaps we should construct the new `dependencies` as we are constructing `rv`.